From: NIIBE Yutaka <gniibe@fsij.org>
Date: Thu, 26 Nov 2020 09:50:40 +0900
Subject: gpg: Report an error for receiving key from agent.

* g10/export.c (do_export_one_keyblock): Report an error.

--

GnuPG-bug-id: 5151
Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
(cherry picked from commit 6f0066db2c87e6362473d17c0621011ed1e1eae6)
---
 g10/export.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/g10/export.c b/g10/export.c
index e98af59..8e17df3 100644
--- a/g10/export.c
+++ b/g10/export.c
@@ -1814,6 +1814,7 @@ do_export_one_keyblock (ctrl_t ctrl, kbnode_t keyblock, u32 *keyid,
                 {
                   if (gpg_err_code (err) == GPG_ERR_FULLY_CANCELED)
                     goto leave;
+                  write_status_error ("export_keys.secret", err);
                   skip_until_subkey = 1;
                   err = 0;
                 }
